### DISCUSSION TITLE
Add more space to a room (S55)

### DISCUSSION THREAD

**Discussion by ottelo9** [rand256/valetudo#434] [Q&A]
I have few rooms on my map. One room (HWR) had a hidden/blocked space inside, because of a big object. I now removed the object and start a room cleanup. Now the previously blocked space is bright blue BUT its not inside the "HWR" room because the color is not white (see image). If I start a room cleanup the roborock will clean around the space. How can I add the tiny space to the "HWR" room?
![2021_05_27_21_33_48_Overview_Home_Assistant](https://user-images.githubusercontent.com/33122175/119886511-57128180-bf33-11eb-8e06-c10af102a6a7.png)

**Comment by rand256** [comment]
Unfortunately there are no MiIO commands I'm aware of that could easily fix such issues. You could probably ask the device to perform zoned cleaning or local cleaning of the new space (or just move it there) and hope it would append the new area to some of existing rooms.

But sometimes even that could fail. I have kinda similar bug that after moving some furniture out of the room, the device now thinks that the new area in the room belongs to another room that is even NOT directly connected to the first one. So I can't just cut and merge it correctly but rather have to re-merge and split from scratch almost all the house to get it fixed. This is one of low-level firmware faults I believe...
